### ISSUE TITLE
Evaluate effort required to incorporate non-gen AI quotes into article summaries

### ISSUE THREAD

**Issue by ndbolligerD3** [harvard-hbs-d3/d3-open-webui#9]

**Comment by tflinthbs** [comment]
@santigamo copying your question here cc: @ndbolligerD3 
> Are you specifically referring to exploring something similar to Retrieval Interleaved Generation (RIG)? If so, could you provide more context or examples?

Yes, certainly Retrieval Interleaved Generation is one of the approaches that could solve this problem. Here's an example of the type of quotes that we're constantly looking to pull from research papers. And they do have to be 100% accurate. 

The last tab in the spreadsheet is an example of quotes pulled from event transcripts and internet and that was done manually. 

These facts are used in our presentations and promotional materials.

[State of AI Statements.xlsx](https://github.com/user-attachments/files/18102399/State.of.AI.Statements.xlsx)
(I'll upload a clean version of this to an S3 bucket including true/false facts that we can potentially use for more automated testing.)

**Comment by Elmolesto** [comment]
@tflinthbs 
I was looking into the RIG topic, and it's definitely beyond our reach. It's not just about Open Web UI, but rather that this is a technique Google is still in the early stages of researching.
It requires fine-tuning a model, and the published version only works with Google’s Data Commons dataset and is not usable or adaptable for other use cases.

I would suggest exploring alternative techniques within RAG to improve citation accuracy and incorporating a mechanism to validate them.

To continue working on this, would be very helpful to have a few test cases to work with.

CC @ndbolligerD3

**Comment by ndbolligerD3** [comment]
@Elmolesto Here are a couple of cleaned up test cases. I'll add more as we go. I tested these on dev with the appropriately uploaded research papers.[Test Suite for IPI accuracy(Sheet1).csv](https://github.com/user-attachments/files/18200038/Test.Suite.for.IPI.accuracy.Sheet1.csv)

**Comment by ndbolligerD3** [comment]
I only did successful ones so we could have a baseline. I can add ones that failed in Open WebUI and succeeded in IPI v1.

**Comment by Elmolesto** [comment]
@ndbolligerD3 Yes, that would be ideal—to know the failing cases so we can address and correct them.
